openfisca_core/periods/helpers.py

```diff
@@ -162,15 +180,9 @@ def period(value) -> Period:
         return Period((DateUnit.DAY, instant(value), 1))
 
     # We return an "eternity-period", for example
-    # ``<Period(('eternity', <Instant(1, 1, 1)>, inf))>``.
+    # ``<Period(('eternity', <Instant(1, 1, 1)>, 0))>``.
```

Comment:
Are we sure we do not break anything with this change ?

Reply:
It may happen, but it would be strange, as it will imply that you handle by default periods as floats and not as integers (integers are a subtype of floats). If that was the case, it would be a waste of memory, you really just need ints here. Should I mark it as a major bump and announce it?

Reply:
I think so.

Reply:
The constraint is to make `Instant` a pure `int` construct.  Ideally, to do the same with `Period`, mimicking what is already done with `indexed_enums` (that on another PR). Does it make sense?

Reply:
OK